### PR TITLE
Added Web App Example For Using Data Protection Service

### LINKED ISSUE #28765: Data Protection Consumer API - Web App Implementation Missing
Since this is ASP.NET Core documentation, instead of a console app implementation, there should be explanations on how to make use of data protection service in a web app (how to add the service to the services container, how to set up DI in controllers, etc.)

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 23af4f99-387c-b2f5-3d5b-3d98cfac247c
* Version Independent ID: 0d95b7b7-5e46-5718-ca09-bdae43c86963
* Content: [Consumer APIs overview for ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/security/data-protection/consumer-apis/overview?view=aspnetcore-7.0)
* Content Source: [aspnetcore/security/data-protection/consumer-apis/overview.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/security/data-protection/consumer-apis/overview.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

---
[Associated WorkItem - 82760](https://dev.azure.com/msft-skilling/Content/_workitems/edit/82760)

### PR DESCRIPTION
**Pull request by Emrebener**
Fixes #28765

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/security/data-protection/consumer-apis/overview.md](https://github.com/dotnet/AspNetCore.Docs/blob/1f59ab2994a98b8bf2bd0ff49ee85e91caf68a55/aspnetcore/security/data-protection/consumer-apis/overview.md) | [Consumer APIs overview for ASP.NET Core](https://review.learn.microsoft.com/en-us/aspnet/core/security/data-protection/consumer-apis/overview?branch=pr-en-us-28803) |

### CODE REVIEW

File: aspnetcore/security/data-protection/consumer-apis/overview.md [hunk 1]
```diff
@@ -39,9 +39,18 @@ The following sample demonstrates three concepts:
...
 
-The package Microsoft.AspNetCore.DataProtection.Abstractions contains an extension method `IServiceProvider.GetDataProtector` as a developer convenience. It encapsulates as a single operation both retrieving an `IDataProtectionProvider` from the service provider and calling `IDataProtectionProvider.CreateProtector`. The following sample demonstrates its usage.
+## Web Application Example
+
+[!code-csharp[](../using-data-protection/samples/webappexample1.cs?highlight=5)]
+
+[!code-csharp[](../using-data-protection/samples/webappexample2.cs?highlight=3,4,5,6,7,8,18,22)]
+
+> [!NOTE]
+> The package Microsoft.AspNetCore.DataProtection.Abstractions contains an extension method `IServiceProvider.GetDataProtector` as a developer convenience. It encapsulates as a single operation both retrieving an `IDataProtectionProvider` from the service provider and calling `IDataProtectionProvider.CreateProtector`. The following sample demonstrates its usage.
```
Comment (Rick-Anderson) [inline aspnetcore/security/data-protection/consumer-apis/overview.md:53]:
```suggestion
The package `Microsoft.AspNetCore.DataProtection.Abstractions` contains an extension method `IServiceProvider.GetDataProtector` as a developer convenience. It encapsulates as a single operation both retrieving an `IDataProtectionProvider` from the service provider and calling `IDataProtectionProvider.CreateProtector`. The following sample demonstrates its usage:
```

File: aspnetcore/security/data-protection/consumer-apis/overview.md [hunk 2]
```diff
@@ -39,9 +39,18 @@ The following sample demonstrates three concepts:
...
+## Console Application Example
+
 [!code-csharp[](../using-data-protection/samples/protectunprotect.cs?highlight=26,34,35,36,37,38,39,40)]
 
-The package Microsoft.AspNetCore.DataProtection.Abstractions contains an extension method `IServiceProvider.GetDataProtector` as a developer convenience. It encapsulates as a single operation both retrieving an `IDataProtectionProvider` from the service provider and calling `IDataProtectionProvider.CreateProtector`. The following sample demonstrates its usage.
+## Web Application Example
+
+[!code-csharp[](../using-data-protection/samples/webappexample1.cs?highlight=5)]
+
+[!code-csharp[](../using-data-protection/samples/webappexample2.cs?highlight=3,4,5,6,7,8,18,22)]
```
Comment (Rick-Anderson) [inline aspnetcore/security/data-protection/consumer-apis/overview.md:50]:
```suggestion
[!code-csharp[](~/security/data-protection/using-data-protection/samples/webappexample1.cs?highlight=5)]

[!code-csharp[](~/security/data-protection/using-data-protection/samples/webappexample2.cs?highlight=3,4,5,6,7,8,18,22)]
```

File: aspnetcore/security/data-protection/consumer-apis/overview.md [hunk 3]
```diff
@@ -39,9 +39,17 @@ The following sample demonstrates three concepts:
...
 [!code-csharp[](../using-data-protection/samples/protectunprotect.cs?highlight=26,34,35,36,37,38,39,40)]
 
-The package Microsoft.AspNetCore.DataProtection.Abstractions contains an extension method `IServiceProvider.GetDataProtector` as a developer convenience. It encapsulates as a single operation both retrieving an `IDataProtectionProvider` from the service provider and calling `IDataProtectionProvider.CreateProtector`. The following sample demonstrates its usage.
+## Web Application Example
+
+[!code-csharp[](~/security/data-protection/using-data-protection/samples/webappexample1.cs?highlight=5)]
+
+[!code-csharp[](~/security/data-protection/using-data-protection/samples/webappexample2.cs?highlight=3,4,5,6,7,8,18,22)]
+
+The package `Microsoft.AspNetCore.DataProtection.Abstractions` contains an extension method `IServiceProvider.GetDataProtector` as a developer convenience. It encapsulates as a single operation both retrieving an `IDataProtectionProvider` from the service provider and calling `IDataProtectionProvider.CreateProtector`. The following sample demonstrates its usage:
```
Comment (Rick-Anderson) [inline aspnetcore/security/data-protection/consumer-apis/overview.md:52]:
```suggestion
The package `Microsoft.AspNetCore.DataProtection.Abstractions` contains an extension method <xref:Microsoft.AspNetCore.DataProtection.DataProtectionCommonExtensions.GetDataProtector%2A> as a developer convenience. It encapsulates as a single operation both retrieving an <xref:Microsoft.AspNetCore.DataProtection.IDataProtectionProvider> from the service provider and calling [`IDataProtectionProvider.CreateProtector`](xref:Microsoft.AspNetCore.DataProtection.IDataProtectionProvider.CreateProtector%2A). The following sample demonstrates its usage:
```

File: aspnetcore/security/data-protection/consumer-apis/overview.md [hunk 4]
```diff
@@ -39,9 +39,17 @@ The following sample demonstrates three concepts:
 
 3. Creating an `IDataProtector` from an `IDataProtectionProvider` and using it to protect and unprotect data.
 
+## Console Application Example
+
 [!code-csharp[](../using-data-protection/samples/protectunprotect.cs?highlight=26,34,35,36,37,38,39,40)]
 
-The package Microsoft.AspNetCore.DataProtection.Abstractions contains an extension method `IServiceProvider.GetDataProtector` as a developer convenience. It encapsulates as a single operation both retrieving an `IDataProtectionProvider` from the service provider and calling `IDataProtectionProvider.CreateProtector`. The following sample demonstrates its usage.
+## Web Application Example
```
Comment (Rick-Anderson) [inline aspnetcore/security/data-protection/consumer-apis/overview.md:46]:
```suggestion
## Web app
```

File: aspnetcore/security/data-protection/consumer-apis/overview.md [hunk 5]
```diff
@@ -39,9 +39,17 @@ The following sample demonstrates three concepts:
 
 3. Creating an `IDataProtector` from an `IDataProtectionProvider` and using it to protect and unprotect data.
 
+## Console Application Example
```
Comment (Rick-Anderson) [inline aspnetcore/security/data-protection/consumer-apis/overview.md:42]:
```suggestion
## Console app
```

File: aspnetcore/security/data-protection/consumer-apis/overview.md [hunk 6]
```diff
@@ -39,9 +39,17 @@ The following sample demonstrates three concepts:
...
 
+## Console app
+
 [!code-csharp[](../using-data-protection/samples/protectunprotect.cs?highlight=26,34,35,36,37,38,39,40)]
 
-The package Microsoft.AspNetCore.DataProtection.Abstractions contains an extension method `IServiceProvider.GetDataProtector` as a developer convenience. It encapsulates as a single operation both retrieving an `IDataProtectionProvider` from the service provider and calling `IDataProtectionProvider.CreateProtector`. The following sample demonstrates its usage.
+## Web app
+
+[!code-csharp[](~/security/data-protection/using-data-protection/samples/webappexample1.cs?highlight=5)]
+
```
Comment (Rick-Anderson) [inline aspnetcore/security/data-protection/consumer-apis/overview.md:49]:
```suggestion

The following highlighted code shows how to use <xref:Microsoft.AspNetCore.DataProtection.IDataProtector> in a controller:

```

File: aspnetcore/security/data-protection/consumer-apis/overview.md [hunk 7]
```diff
@@ -39,9 +39,17 @@ The following sample demonstrates three concepts:
 
 3. Creating an `IDataProtector` from an `IDataProtectionProvider` and using it to protect and unprotect data.
 
+## Console app
+
 [!code-csharp[](../using-data-protection/samples/protectunprotect.cs?highlight=26,34,35,36,37,38,39,40)]
 
-The package Microsoft.AspNetCore.DataProtection.Abstractions contains an extension method `IServiceProvider.GetDataProtector` as a developer convenience. It encapsulates as a single operation both retrieving an `IDataProtectionProvider` from the service provider and calling `IDataProtectionProvider.CreateProtector`. The following sample demonstrates its usage.
+## Web app
+
```
Comment (Rick-Anderson) [inline aspnetcore/security/data-protection/consumer-apis/overview.md:47]:
```suggestion
Call <xref:Microsoft.Extensions.DependencyInjection.DataProtectionServiceCollectionExtensions.AddDataProtection(Microsoft.Extensions.DependencyInjection.IServiceCollection,System.Action{Microsoft.AspNetCore.DataProtection.DataProtectionOptions})> in `Program.cs`:

```

**Comment by Rick-Anderson** [comment]
@blowdart  please review

**Review by blowdart — APPROVED** [review]

**Comment by Rick-Anderson** [comment]
Thanks for the contribution. Be sure to check out your name as a contributor about the 3rd of next month on the [What's new page](https://docs.microsoft.com/en-us/aspnet/core/whats-new/) in the **Community contributors** section at the bottom.

Check out [is:issue is:open label:"code help wanted" "](https://github.com/dotnet/AspNetCore.Docs/issues?q=is%3Aissue+is%3Aopen+label%3A%22code+help+wanted%22+) and [is:open is:issue label:"Good first issue"](https://github.com/dotnet/AspNetCore.Docs/issues?q=is%3Aopen+is%3Aissue+label%3A%22Good+first+issue%22)

**Pull request merged**